config-model/src/main/java/com/yahoo/vespa/model/container/xml/ContainerModelBuilder.java

```diff
@@ -1034,7 +1035,20 @@ private ZoneEndpoint zoneEndpoint(ConfigModelContext context, ClusterSpec.Id clu
         InstanceName instance = context.properties().applicationId().instance();
         ZoneId zone = ZoneId.from(context.properties().zone().environment(),
                                   context.properties().zone().region());
-        return context.getApplicationPackage().getDeploymentSpec().zoneEndpoint(instance, zone, cluster);
+
+        var supportsTokenAuthentication = context.properties()
+                .endpoints()
+                .stream()
+                .anyMatch(endpoint -> endpoint.authMethod() == ApplicationClusterEndpoint.AuthMethod.token);
```

Comment:
You also need to match the cluster here, since this is the zone endpoint for a particular cluster and auth method is a per-cluster setting.